pkg/raft/raft.go

```diff
@@ -542,6 +542,10 @@ func (r *raft) send(m pb.Message) {
 			//   same reasons MsgPreVote is
 			r.logger.Panicf("term should be set when sending %s", m.Type)
 		}
+	case pb.MsgApp:
```

Comment:
is this just a missing safety check that you're adding in this commit or is there a closer relation? If not, might be worth extracting so that you get a chance to explain what was missing and what the impact could've been.